implementations/cpp-jsontoolkit/bowtie_jsontoolkit.cpp

```diff
@@ -0,0 +1,113 @@
+#include <sourcemeta/jsontoolkit/json.h>
+#include <sourcemeta/jsontoolkit/jsonschema.h>
+
+#include <cassert>
+#include <cstdlib>
+#include <iostream>
+#include <optional>
+#include <string>
+#include <utility>
+
+int main() {
+  using namespace sourcemeta::jsontoolkit;
+  bool started{false};
+  std::optional<std::string> default_dialect{std::nullopt};
+
+  std::string line;
+  while (std::getline(std::cin, line)) {
+    const JSON message{parse(line)};
+
+    assert(message.defines("cmd"));
+    assert(message.at("cmd").is_string());
+
+    const std::string command{message.at("cmd").to_string()};
+
+    if (command == "start") {
+      started = true;
+      assert(message.defines("version") && message.at("version").is_integer());
+      assert(message.at("version").to_integer() == 1);
+
+      auto response{JSON::make_object()};
+      response.assign("version", JSON{1});
+      auto implementation{JSON::make_object()};
+      implementation.assign("language", JSON{"cpp"});
+      implementation.assign("name", JSON{"jsontoolkit"});
+      implementation.assign("homepage",
+                            JSON{"https://github.com/sourcemeta/jsontoolkit"});
+      implementation.assign(
+          "issues", JSON{"https://github.com/sourcemeta/jsontoolkit/issues"});
+      implementation.assign("source",
+                            JSON{"https://github.com/sourcemeta/jsontoolkit"});
+      implementation.assign(
+          "dialects", JSON{JSON{"http://json-schema.org/draft-07/schema#"},
+                           JSON{"http://json-schema.org/draft-06/schema#"},
+                           JSON{"http://json-schema.org/draft-04/schema#"}});
+
+      response.assign("implementation", std::move(implementation));
+      stringify(response, std::cout);
+      std::cout << std::endl;
+    } else if (command == "dialect") {
+      assert(started);
+      assert(message.defines("dialect") && message.at("dialect").is_string());
+      default_dialect = message.at("dialect").to_string();
+      auto response{JSON::make_object()};
+      response.assign("ok", JSON{true});
+      stringify(response, std::cout);
+      std::cout << std::endl;
+    } else if (command == "run") {
+      assert(started);
+      assert(message.defines("seq") && message.at("seq").is_integer());
```

Comment:
It's actually not an integer! `bowtie smoke` (which I'm in the middle of beefing up) will soon warn harness authors of that automatically, but yeah `seq` [has no public type](https://github.com/bowtie-json-schema/bowtie/blob/main/bowtie/schemas/report.json#L25-L28), it's an opaque JSON value you're supposed to pass along (and in fact for internal transition reasons it's not an integer when running `bowtie suite`).

Reply:
Ahhh, got it. Might be worth explaining just that on the "new implementation" guide. Let me fix it now...